src/ua_types.c

```diff
@@ -1010,13 +1011,37 @@ copyStructure(const void *src, void *dst, const UA_DataType *type) {
     for(size_t i = 0; i < type->membersSize; ++i) {
         const UA_DataTypeMember *m= &type->members[i];
         const UA_DataType *mt = &typelists[!m->namespaceZero][m->memberTypeIndex];
-        if(!m->isArray) {
+        if (m->isOptional){
```

Comment:
The compiler makes a "fast path" that has minimum jump instructions.
(Exceptions to the rule exist of course.)

A standalone `if` has a jump-instruction.
For an `if-else`, the first if does not have jump instructions.

This is speed-critical code.
Try to make the most common operations go in the first clause of the `if-else`.

If that makes the code ugly, we can add annotation to help the branch prediction.
But we haven't done this in the past.

```c
#define likely(x)       __builtin_expect((x),1)
#define unlikely(x)     __builtin_expect((x),0)
```